src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+    private double maxAbnormalFraction;
+    private double maxCopyNumber;
+    private double minHetAlleleFraction;
+    private double breakpointHetAlleleFraction;
+
+    /* integration quadrature */
+    @VisibleForTesting
+    final ArrayList<Double> gaussIntegrationWeights = new ArrayList<>();
+    @VisibleForTesting
+    final ArrayList<Double> gaussIntegrationLogWeights = new ArrayList<>();
+    @VisibleForTesting
+    final ArrayList<Double> gaussIntegrationAbscissas = new ArrayList<>();
+
+    /* allele fraction prior for Het sites */
+    @VisibleForTesting
+    final ArrayList<Double> alleleFractionPriors = new ArrayList<>();
+    @VisibleForTesting
+    final ArrayList<Double> alleleFractionLogPriors = new ArrayList<>();
+
+    /* minimum order of the integration quadrature */
+    private static final int MIN_QUADRATURE_ORDER = 50;
+
+    /* interval threshold for indexing for SamLocusIterator */
+    private static final int MAX_INTERVALS_FOR_INDEX = 25000;
+
+    /* default priors */
+    private static final double DEFAULT_PRIOR_REF_HOM = 0.5; /* a homozygous site being the ref allele */
+    private static final double DEFAULT_PRIOR_HET = 0.5; /* a site being heterozygous */
+
+    /* a third of the minimum sequencing error (for safeguarding log likelihood calculations) */
+    private static final double DEFAULT_MIN_BASE_ERROR_THIRD = 1e-6;
+
+    /* approximate number of status updates printed to log */
+    private static final int NUMBER_OF_SITES_PER_LOGGED_STATUS_UPDATE = 10000;
+
+    /**
+     * Constructor of {@link BayesianHetPulldownCalculator} object
+     *
+     * NOTE: The default {@link HetPriorType} is BALANCED. Make a call to
+     *       {@link BayesianHetPulldownCalculator#useHeterogeneousHetPrior} to switch to the HETEROGENEOUS prior.
+     *
+     * @param refFile the reference genome file
+     * @param snpIntervals {@link IntervalList} of common SNPs
+     * @param minMappingQuality minimum phred mapping quality
+     * @param minBaseQuality minimum phred base quality
+     * @param readDepthThreshold minimum read depth
+     * @param validationStringency validation stringency
+     * @param errorProbabilityAdjustmentFactor (experimental) multiplicative factor for read and mapping error
+     *                                         probabilities
+     */
+    public BayesianHetPulldownCalculator(final File refFile, final IntervalList snpIntervals,
+                                         final int minMappingQuality, final int minBaseQuality,
+                                         final int readDepthThreshold, final ValidationStringency validationStringency,
+                                         final double errorProbabilityAdjustmentFactor) {
+
+        ParamUtils.isPositiveOrZero(minMappingQuality, "Minimum mapping quality must be nonnegative.");
+        ParamUtils.isPositiveOrZero(minBaseQuality, "Minimum base quality must be nonnegative.");
+
+        this.refFile = refFile;
```

Comment:
Is ```refFile``` allowed to be ```null```?  If not due ```this.refFile  = Utils.nonNull(refFile)```
Same for ```snpIntervals```. 

Also you need to check that the input snpIntervals are legal snp intervals (eg. no nulls and 1 bp in length).